### 关于deltamotion_cloud数据迁移过程中冲突的问题解决

#### 1. 原因

在中间的某个commit，将service模块中已存在的model迁移到了identity模块，在两个模块下均生成了对应的migrations版本文件；在这种情况下，发布该commit的过程中，migrate时有顺序的要求，即需要先对service模块进行migrate，再对identity模块进行migrate。而在正常重新搭建项目及数据库时，migrate的过程是按照模块的顺序执行各模块的migrattions文件，identity模块的顺序较service模块优先，所以会出现migrate失败的问题。

#### 2. 解决方案

根据1.中原因，针对该commit进行手动发布，调整service模块和identity模块的migrate顺序，以此来解决该问题。

#### 3. 具体操作步骤

```bash
# 1.拉取代码
git pull origin dev

# 2.回退至出现问题commit版本(41ef44bc)的上一个版本(d89526c4)
git reset --hard d89526c4

# 3.进入到server容器中，进行数据迁移migrate
docker-compose exec server bash
python manage.py migrate

# 4.拉取代码，回退至出现问题的commit版本(41ef44bc)
git pull origin dev
git reset --hard 41ef44bc

# 5.进行数据迁移操作，注意迁移的顺序
docker-compose exec server bash
python manage.py migrate service
python manage.py migrate identity

# 6.拉取最新代码，进行数据迁移
git pull origin dev
docker-compose exec server bash
python manage.py migrate
```

